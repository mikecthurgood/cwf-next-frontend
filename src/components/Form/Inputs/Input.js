import React from 'react';
import Styled from 'styled-components';

const input = props => (
  <InputContainer compact={props.compact} >
    {props.label && <label htmlFor={props.id}>{props.label}</label>}
    <input
      className={[
        !props.valid ? 'invalid' : 'valid',
        props.touched ? 'touched' : 'untouched',
        props.loginError ? 'errorFlash' : ''
      ].join(' ')}
      type={props.type}
      id={props.id}
      required={props.required}
      value={props.value}
      placeholder={props.placeholder}
      onChange={e => props.onChange(props.id, e.target.value, e.target.files)}
      onBlur={props.onBlur}
    />
  </InputContainer>
);

export default input;

const InputContainer = Styled.div`
    margin: ${({compact}) => compact ? '0.4rem 0' : '1rem 0'};
    width: 100%;
    overflow: hidden;

  label {
    display: block;
    text-transform: uppercase;
    margin-bottom: 0.25rem;
    color: #282c34;
  }

  input {
    display: block;
    font: inherit;
    padding: 0.25rem 0.5rem;
    width: 100%;
    border-radius: 3px;
    border: 1px solid #ccc;
  }

  .touched.invalid {
      border-color: red;
      background: #ffc2c2;
  }

  textarea:focus {
    outline: none;
    border-color: #3b0062;
    color: #3b0062;
  }
  
  .errorFlash {
    position: relative;
    animation: shake .1s linear;
    animation-iteration-count: 3;
    background-color: rgba(255, 44, 44, 0.629);
    
    @keyframes shake {
      0% { left: 0 }
      100% { right: -5px; }
    }
  }
`
