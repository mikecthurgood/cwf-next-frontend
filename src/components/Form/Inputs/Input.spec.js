import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import Input from './Input';

describe('Form Input', () => {
    afterEach(cleanup);
    it('should render the component', () => {
        const props = {
            valid: false,
            touched: false,
            loginError: false,
        }
        const { container } = render(
            <Input {...props} />
        );

        expect(container).toMatchSnapshot()
    });

    it('should render the component', () => {
        const props = {
            valid: true,
            touched: true,
            loginError: true,
            label: 'test label',
        }
        const { container } = render(
            <Input {...props} />
        );
        expect(container).toMatchSnapshot()
    });

    it('should render the component', () => {
        const props = {
            valid: true,
            touched: true,
            loginError: true,
            label: 'test label',
            onChange: jest.fn(),
        }
        const { getByTestId } = render(
            <Input {...props} />
        );

        fireEvent.change(getByTestId('form-input'), { target: { value: 'Good Day' } })
        expect(props.onChange).toBeCalled();
        
    })
});
