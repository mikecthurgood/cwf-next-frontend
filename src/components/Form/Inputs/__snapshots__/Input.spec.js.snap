// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input should render the component 1`] = `
<div>
  <div
    class="Input__InputContainer-sc-1wyp7tv-0 kIYPqy"
  >
    <input
      class="invalid untouched "
      data-testid="form-input"
      value=""
    />
  </div>
</div>
`;

exports[`Form Input should render the component 2`] = `
<div>
  <div
    class="Input__InputContainer-sc-1wyp7tv-0 kIYPqy"
  >
    <label>
      test label
    </label>
    <input
      class="valid touched errorFlash"
      data-testid="form-input"
      value=""
    />
  </div>
</div>
`;
